File: 백준/Bronze/5073. 삼각형과 세 변/삼각형과 세 변.py
while True:
    shape = list(map(int, input().split()))
    shape.sort()
    a,b,c = shape[0],shape[1],shape[2]
    if a==0 and b==0 and c==0:
        break

    if c >= a+b:
        print("Invalid")

    elif a==b and b==c and c==a:
        print("Equilateral")
    elif a != b and b!= c and a!=c:
        print("Scalene")
    else:
        print("Isosceles")

File: 백준/Bronze/5073. 삼각형과 세 변/삼각형과 세 변.cc
#include <iostream>
#include <vector>
#include <algorithm>

using namespace std;

const int INF = 100001;

int main() {
    ios_base::sync_with_stdio(false);
	cout.tie(NULL);
	cin.tie(NULL);
	
	int a,b,c;
	string ans;
	while(cin >> a >> b >> c) {
	    if(a==0 && b==0 && c==0) {
	        break;
	    }
	    
	    if(!((a<b+c) && (b<a+c) && (c<a+b))) {
	        ans = "Invalid";
	    } else if(a==b && b==c) {
	        ans = "Equilateral";
	    } else if(a==b || b==c || a==c) {
	        ans = "Isosceles";
	    } else {
	        ans = "Scalene";
	    }
	    
	    cout << ans << "\n";
	}

    return 0;
}